agent/src/agent/pipeline/config/js_scripts/protocol_3.js

```diff
@@ -19,17 +19,13 @@ function replace_illegal_chars(str) {
 
 function get_dimensions(record) {
     var dimensions = {};
-    for (var j = 0; j < state['DIMENSIONS'].length; j++) {
-        var dimension = extract_value(record.value, replace_illegal_chars(state['DIMENSIONS'][j]))
-        if (dimension === null) {
-            continue;
-        }
-        dimension = String(dimension).trim()
-        if (dimension === '') {
-            continue;
+    for (var dimension_path in state['DIMENSIONS']) {
+        var dimension_value = extract_value(record.value, replace_illegal_chars(dimension_path))
+        dimension_value = String(dimension_value).trim()
+        if (dimension_value !== '' && dimension_value !== 'null') {
```

Comment:
revert to the original logic because value can be a 'null' string and it's valid

Reply:
but logic didn't change, I just removed the continue operator and put code into the if block